### PR TITLE
docs: update whatsnew and changelog for 1220 release

### PR DESCRIPTION
**Pull request by MuyangAmigo**
ms-teams-vscode-extension	v4.2.0-rc.2

CLI BREAKING CHANGE:
  

Extension-toolkit BREAKING CHANGE:
 

CLI feat commits:
  

Extension-toolkit feat commits:

### CODE REVIEW

**Review by summzhan — DISMISSED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

File: packages/vscode-extension/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Changelog
 
+## 4.2.0 - Dec 20, 2022
+
+New features:
+
+- Use the new Dashboard tab template to create a personal and collaborative canvas in Teams. Learn more about this new template in the [documentation](https://aka.ms/teamsfx-dashboard-app).
+
+Enhancements:
+- Step-by-step guidance in Visual Studio Code to assisit you while creating a free Microsoft 365 testing tenant.
```
Comment (yiqing-zhao) [inline packages/vscode-extension/CHANGELOG.md:10]: Looks like there is a spelling error of `assist`.

Comment (pierceboggan) [reply packages/vscode-extension/CHANGELOG.md:10]: Would it be possible to link to a screenshot of the dashboard template under our description?

Comment (pierceboggan) [reply packages/vscode-extension/CHANGELOG.md:10]: We had also talked about Dashboard being a sample and not a template, but it seems like perhaps the ship has sailed on that?

File: packages/vscode-extension/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+- SPFx based tab app is updated to use the latest [SharePoint Framework v1.16](https://learn.microsoft.com/sharepoint/dev/spfx/release-1.16).
```
Comment (pierceboggan) [inline packages/vscode-extension/CHANGELOG.md:11]: "Scaffolds that use SharePoint Framework (SPFx) have been updated to use [SharePoint Framework version 1.16](https://learn.microsoft.com/sharepoint/dev/spfx/release-1.16) which introduces the ability to build experiences for Outlook and Office.com (Microsoft 365 app) in addition to Teams.

Comment (MuyangAmigo) [reply packages/vscode-extension/CHANGELOG.md]: Updated

File: packages/vscode-extension/CHANGELOG.md (hunk 1 above)
Comment (MuyangAmigo) [reply packages/vscode-extension/CHANGELOG.md]: Updated screenshot and fixed typo.

**Pull request merged**